ludwig/data/split.py

```diff
@@ -98,6 +99,25 @@ def required_columns(self) -> List[str]:
         return [self.column]
 
 
+def _get_partition_splitter(splitter: Splitter, random_seed: float) -> Callable:
```

Comment:
The function from `_get_partition_splitter` takes the non-partitioned code path (since it is using the local backend). `StratifySplitter.split` already does the correct thing for single partitions, so applying the function from `_get_partition_splitter` to every partition will correctly split each partition using the local backend.

Reply:
Got it, thanks for clarifying!

One possible way of doing this that wouldn't require initializing a local backend, and would make the code clearer is the following:

```
def split_dataframe(df, column, probabilities, random_seed) -> df_train, df_val, df_test:
    ...


class StratifySplitter
    ...

    def split(self, ...):
        if not backend.df_engine.partitioned:
            return split_dataframe(df, self.column, self.probabilities, random_split)

        def split_partition(partition):
            _, val, test = split_dataframe(partition, self.column, probabilities, random_seed)
            partition.loc[val.index, TMP_SPLIT_COL] = 1
            partition.loc[test.index, TMP_SPLIT_COL] = 2
            return partition

        df[TMP_SPLIT_COL] = 0
        df = backend.df_engine.map_partitions(df, split_partition, meta=df)
        # Create df_train, df_val, df_test
        ...
        return df_train, df_val, df_test
```

Reply:
That's a lot cleaner. Thanks for the suggestion!